### ISSUE TITLE
Connecting to mosquitto-docker-container via websockets+TLS (wss) is not working with v2.0.5, but work well with v1.6.12

### ISSUE THREAD

**Issue by anencore94** [eclipse/mosquitto#2055]
Hi guys. First of all, thanks for maintaining this nice projects.

Here is my issue.

I've tested mosquitto with docker image in docker hub with following tags:
- eclipse-mosquitto:1.6.12-openssl
- eclipse-mosquitto:2.0.5-openssl

```
my env
Docker version is: Docker version 19.03.6, build 369ce74a3c
Ubuntu version is: 18.04.
```

I guess this is a similar issue with #2050. But I'm not sure.

The problem is as follows:

1) I've run the same command with the same mosquitto.conf file and same crt, key files.
`sudo docker run -it -p 1883:1883 -p 8883:8883 -v /home/kjy/temp/mosquitto/mosquitto-1.4.11/mosquitto.conf:/mosquitto/config/mosquitto.conf -v /etc/mosquitto/certs:/etc/mosquitto/certs --name msq eclipse-mosquitto:1.6.12-openssl`
VERSUS
`
sudo docker run -it -p 1883:1883 -p 8883:8883 -v /home/kjy/temp/mosquitto/mosquitto-1.4.11/mosquitto.conf:/mosquitto/config/mosquitto.conf -v /etc/mosquitto/certs:/etc/mosquitto/certs --name msq eclipse-mosquitto:2.0.5-openssl
`

2) The response from mosquitto broker is not same, even if the mqtt publisher was same
- the log from mqtt broker (log_type all and websockets_log_level 255) was different:

- v1.6.12 (works well with both wss://localhost:8883/mqtt & mqtt localhost:1883)
```
1611824006: mosquitto version 1.6.12 starting
1611824006: Config loaded from /mosquitto/config/mosquitto.conf.
1611824006: Opening websockets listen socket on port 8883.
1611824006: libuv support not compiled in
1611824006: Creating Vhost 'default' port 8883, 3 protocols, IPv6 on
1611824006:  Using SSL mode
1611824006:  SSL ECDH curve 'prime256v1'
1611824006: Opening ipv4 listen socket on port 1883.
1611824006: Opening ipv6 listen socket on port 1883.
1611824006: mosquitto version 1.6.12 running
1611824006: New connection from 172.17.0.1 on port 1883.
1611824006: New client connected from 172.17.0.1 as 0b27dcb6-cd12-4f5e-8f86-3496d21c7f321611823927518 (p2, c1, k10).
1611824006: No will message specified.
1611824006: Sending CONNACK to 0b27dcb6-cd12-4f5e-8f86-3496d21c7f321611823927518 (0, 0)
1611824007: 
1611824007: 0000: 47 45 54 20 2F 6D 71 74 74 20 48 54 54 50 2F 31    GET /mqtt HTTP/1
1611824007: 0010: 2E 31 0D 0A 43 6F 6E 6E 65 63 74 69 6F 6E 3A 20    .1..Connection: 
1611824007: 0020: 55 70 67 72 61 64 65 0D 0A 55 70 67 72 61 64 65    Upgrade..Upgrade
1611824007: 0030: 3A 20 77 65 62 73 6F 63 6B 65 74 0D 0A 48 6F 73    : websocket..Hos
1611824007: 0040: 74 3A 20 6C 6F 63 61 6C 68 6F 73 74 3A 38 38 38    t: localhost:888
1611824007: 0050: 33 0D 0A 53 65 63 2D 57 65 62 53 6F 63 6B 65 74    3..Sec-WebSocket
1611824007: 0060: 2D 56 65 72 73 69 6F 6E 3A 20 31 33 0D 0A 53 65    -Version: 13..Se
1611824007: 0070: 63 2D 57 65 62 53 6F 63 6B 65 74 2D 4B 65 79 3A    c-WebSocket-Key:
1611824007: 0080: 20 4D 54 4D 74 4D 54 59 78 4D 54 67 79 4E 44 41     MTMtMTYxMTgyNDA
1611824007: 0090: 77 4E 6A 6B 35 4E 77 3D 3D 0D 0A 53 65 63 2D 57    wNjk5Nw==..Sec-W
1611824007: 00A0: 65 62 53 6F 63 6B 65 74 2D 50 72 6F 74 6F 63 6F    ebSocket-Protoco
1611824007: 00B0: 6C 3A 20 6D 71 74 74 0D 0A 53 65 63 2D 57 65 62    l: mqtt..Sec-Web
1611824007: 00C0: 53 6F 63 6B 65 74 2D 45 78 74 65 6E 73 69 6F 6E    Socket-Extension
1611824007: 00D0: 73 3A 20 70 65 72 6D 65 73 73 61 67 65 2D 64 65    s: permessage-de
1611824007: 00E0: 66 6C 61 74 65 3B 20 63 6C 69 65 6E 74 5F 6D 61    flate; client_ma
1611824007: 00F0: 78 5F 77 69 6E 64 6F 77 5F 62 69 74 73 0D 0A 0D    x_window_bits...
1611824007: 0100: 0A                                                 .               
1611824007: 
1611824007: 
1611824007: 0000: 47 45 54 20 2F 6D 71 74 74 20 48 54 54 50 2F 31    GET /mqtt HTTP/1
1611824007: 0010: 2E 31 0D 0A 43 6F 6E 6E 65 63 74 69 6F 6E 3A 20    .1..Connection: 
1611824007: 0020: 55 70 67 72 61 64 65 0D 0A 55 70 67 72 61 64 65    Upgrade..Upgrade
1611824007: 0030: 3A 20 77 65 62 73 6F 63 6B 65 74 0D 0A 48 6F 73    : websocket..Hos
1611824007: 0040: 74 3A 20 6C 6F 63 61 6C 68 6F 73 74 3A 38 38 38    t: localhost:888
1611824007: 0050: 33 0D 0A 53 65 63 2D 57 65 62 53 6F 63 6B 65 74    3..Sec-WebSocket
1611824007: 0060: 2D 56 65 72 73 69 6F 6E 3A 20 31 33 0D 0A 53 65    -Version: 13..Se
1611824007: 0070: 63 2D 57 65 62 53 6F 63 6B 65 74 2D 4B 65 79 3A    c-WebSocket-Key:
1611824007: 0080: 20 4D 54 4D 74 4D 54 59 78 4D 54 67 79 4E 44 41     MTMtMTYxMTgyNDA
1611824007: 0090: 77 4E 6A 6B 35 4E 77 3D 3D 0D 0A 53 65 63 2D 57    wNjk5Nw==..Sec-W
1611824007: 00A0: 65 62 53 6F 63 6B 65 74 2D 50 72 6F 74 6F 63 6F    ebSocket-Protoco
1611824007: 00B0: 6C 3A 20 6D 71 74 74 0D 0A 53 65 63 2D 57 65 62    l: mqtt..Sec-Web
1611824007: 00C0: 53 6F 63 6B 65 74 2D 45 78 74 65 6E 73 69 6F 6E    Socket-Extension
1611824007: 00D0: 73 3A 20 70 65 72 6D 65 73 73 61 67 65 2D 64 65    s: permessage-de
1611824007: 00E0: 66 6C 61 74 65 3B 20 63 6C 69 65 6E 74 5F 6D 61    flate; client_ma
1611824007: 00F0: 78 5F 77 69 6E 64 6F 77 5F 62 69 74 73 0D 0A 0D    x_window_bits...
1611824007: 0100: 0A                                                 .               
1611824007: 
1611824008: New client connected from ::ffff:172.17.0.1 as 4131efac-d520-42b3-9ba3-232c0fd5054b1611823927518 (p2, c1, k10, u'localhost').
1611824008: No will message specified.
1611824008: Sending CONNACK to 4131efac-d520-42b3-9ba3-232c0fd5054b1611823927518 (0, 0)
```

- v2.0.5 (works well only mqtt localhost:1883, not wss)
  - At first, the log was saying permission denied, So I change mode to 666 for cert and key files. v1.6.12 didn't require such permission though.
  - Logs below was happend after I change mode to 666 and re-run docker container.
```
1611823936: The 'port' option is now deprecated and will be removed in a future version. Please use 'listener' instead.
1611823936: mosquitto version 2.0.5 starting
1611823936: Config loaded from /mosquitto/config/mosquitto.conf.
1611823936: Opening websockets listen socket on port 8883.
1611823936: libuv support not compiled in
1611823936: Creating Vhost 'default' port 8883, 3 protocols, IPv6 on
1611823936:  Using SSL mode
1611823936:  SSL ECDH curve 'prime256v1'
1611823936: Opening ipv4 listen socket on port 1883.
1611823936: Opening ipv6 listen socket on port 1883.
1611823936: mosquitto version 2.0.5 running
1611823937: New connection from 172.17.0.1:58190 on port 1883.
1611823937: New client connected from 172.17.0.1:58190 as 0b27dcb6-cd12-4f5e-8f86-3496d21c7f321611823927518 (p2, c1, k10).
1611823937: No will message specified.
1611823937: Sending CONNACK to 0b27dcb6-cd12-4f5e-8f86-3496d21c7f321611823927518 (0, 0)
1611823937: lws_server_socket_service_ssl: lws_change_pollfd failed
1611823938: lws_server_socket_service_ssl: lws_change_pollfd failed
1611823939: lws_server_socket_service_ssl: lws_change_pollfd failed
```

- just websocket (without tls) also works well in both image.

3. this is my mosquitto.conf
```
log_type all
websockets_log_level 255

port 1883
allow_anonymous true

listener 8883
protocol websockets
cafile /etc/mosquitto/certs/ca.crt
certfile /etc/mosquitto/certs/server.crt
keyfile /etc/mosquitto/certs/server.key
require_certificate true
use_identity_as_username true
```

4. With these logs, First I guess the `libwebsockets` version is different in docker imagefile. BUT I see the lws version on both dockerfiles were same from the source code..
Could you tell me what makes this difference between such two image ?
Is there any way to use wss in mosquitto_v2.0.5 ?

**Comment by ralight** [comment]
This is a problem with 2.0.5, see: https://github.com/eclipse/mosquitto/issues/2050 for example. 2.0.6 will be available today, it will be merged to the new Docker image shortly after.

**Comment by ralight** [comment]
This is now just waiting on docker hub to create the new image, so I'm closing this.

**Comment by anencore94** [comment]
Thanks a lot! I've checked this problem is fixed at v2.0.6.
